pkg/proxy/iptables/proxier.go

```diff
@@ -325,7 +325,7 @@ func NewProxier(ctx context.Context,
 	// We pass syncPeriod to ipt.Monitor, which will call us only if it needs to.
 	// We need to pass *some* maxInterval to NewBoundedFrequencyRunner anyway though.
 	// time.Hour is arbitrary.
-	proxier.syncRunner = async.NewBoundedFrequencyRunner("sync-runner", proxier.syncProxyRules, minSyncPeriod, time.Hour, burstSyncs)
+	proxier.syncRunner = async.NewBoundedFrequencyRunner("sync-runner", proxier.forceSyncProxyRules, minSyncPeriod, time.Hour, burstSyncs)
```

Comment:
Alas, no, it's not that simple; this changes _all_ of the syncs via the BFR to be force syncs. There's (currently) no way to make the BFR do partial syncs normally and full syncs at `maxInterval`.

The easiest fix is probably to have the Proxier keep track of the last time it did a full sync, and then at the top of `syncProxyRules`, when it's deciding whether to `tryPartialSync` or not, it should take into account how long it has been since the last full sync.

Also, probably the `time.Hour` should be assigned to a constant (`requiredFullSyncInterval`? something like that...) rather than being hardcoded.

And yes, we should change each of the proxies to do this. (They all have basically the same architecture so it will be identical in each.) Though maybe that would be a good opportunity for refactoring a little bit? (If not, I have plans to get to this later.)